storage/database.go

```diff
@@ -170,90 +175,78 @@ func (d *db) Close() error {
 	}
 	d.state = databaseClosed
 
-	// For now just remove all shards, in future this could be made more explicit.  However
+	// For now just remove all namespaces, in future this could be made more explicit.  However
 	// this is nice as we do not need to do any other branching now in write/read methods.
-	for i := range d.shards {
-		d.shards[i] = nil
-	}
+	d.namespaces = nil
+
 	for i := 0; i < dbOngoingTasks; i++ {
 		d.doneCh <- struct{}{}
 	}
+
 	// Finally close the commit log
 	return d.commitLog.Close()
 }
 
 func (d *db) Write(
 	ctx context.Context,
+	namespace string,
 	id string,
 	timestamp time.Time,
 	value float64,
 	unit xtime.Unit,
 	annotation []byte,
 ) error {
 	d.RLock()
-	shardID := d.shardSet.Shard(id)
-	shard := d.shards[shardID]
+	n, exists := d.namespaces[namespace]
 	d.RUnlock()
 
-	if shard == nil {
-		return fmt.Errorf("not responsible for shard %d", shardID)
+	if !exists {
+		return fmt.Errorf("not responsible for namespace %s", namespace)
```

Comment:
Perhaps better namespace error would be `fmt.Errorf("no such namespace: %s", namespace)`